sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala

```diff
@@ -630,11 +639,14 @@ abstract class ParquetFilterSuite extends QueryTest with ParquetTest with Shared
         withSQLConf(SQLConf.PARQUET_OUTPUT_TIMESTAMP_TYPE.key ->
           ParquetOutputTimestampType.INT96.toString) {
           import testImplicits._
-          withParquetDataFrame(
-            millisData.map(i => Tuple1(Timestamp.valueOf(i))).toDF()) { implicit df =>
-            val schema = new SparkToParquetSchemaConverter(conf).convert(df.schema)
-            assertResult(None) {
-              createParquetFilters(schema).createFilter(sources.IsNull("_1"))
+          withTempPath { file =>
+            millisData.map(i => Tuple1(Timestamp.valueOf(i))).toDF
+              .write.format(dataSourceName).save(file.getCanonicalPath)
+            readParquetFile(file.getCanonicalPath) { df =>
```

Comment:
From 2 lines to 4 lines? This looks like an exception. Is this inevitable?

Reply:
Yup. I couldn't fine a better way without having another method.

Reply:
Yup. I couldn't find a better way without having another method.